stdlib/deplinker/elf.py

```diff
@@ -0,0 +1,206 @@
+                'bin/*'
+                'usr/bin/*'
+                'lib/*'
+                'usr/lib/*'
+                'lib32/*'
+                'usr/lib32/*'
+                'lib64/*'
+                'usr/lib64/*'
+            ]
+
+    :paramtype search_patterns: ``List`` [ :py:class:`.Package` ]
+    :param local_resolving: Indicate whether or not other packages for ``packages`` should be used to solve dependencies. Default value is ``True``.
+    :param remote_resolving: Indicate whether or not remote repositories should be used to solve dependencies. Default value is ``True``.
+    """
+    binaries = dict()  # Key = a binary filename, Value = a package ID
+    dependencies = dict()  # Key = a package ID, Value = a list of ELF filename
+
+    for package in packages.values():
+        stdlib.log.ilog(f"ELF Dependency Linker: Looking for binaries and their dependencies in {package.id.full_name()}...")
+
+        pkg_id = package.id
+
+        # Find all elfs in this package
+        elfs = _find_elfs(package, search_patterns)
+
+        # Fill the `binaries` dict, to keep track of who owns a binary in the local context
+        #
+        # This generates unspecified behaviour if two packages have two binaries with the same filename
+        for elf in map(ntpath.basename, elfs):
+            binaries[elf] = pkg_id
+
+        # Find the dependencies of every binaries
+        dependencies[pkg_id] = list()
+        for elf in elfs:
+            elf_dependencies = _fetch_elf_dependencies(package, elf)
+            dependencies[pkg_id] += elf_dependencies
+
+            # Log results
+            with stdlib.log.pushlog():
+                stdlib.log.ilog(f"/{elf}: {' '.join(elf_dependencies)}")
+
+        # Remove duplicate
+        dependencies[pkg_id] = list(dict.fromkeys(dependencies[pkg_id]))
+
+    # So from now on, we have all the dependencies in `dependencies`, and all
+    # the binaries in `binaries`.
+    #
+    # We want to try and match the dependencies with the existing binaries,
+    # and if it's not possible, with the remote repositories.
+
+    for (package_id, dependencies) in dependencies.items():
+
+        stdlib.log.ilog(f"ELF Dependency Linker: Solving dependencies of {package_id.full_name()}...")
+        requirements = {}
+
+        with stdlib.log.pushlog():
+
+            for dependency in dependencies:
+                stdlib.log.ilog(f"Looking for a package containing \"{dependency}\"...")
+
+                if local_resolving and dependency in binaries.keys():
+                    dependency_id = binaries[dependency]
+
+                    requirements.update({dependency_id.full_name(): f'={dependency_id.version}'})
+                    stdlib.log.slog(f"Found locally: {dependency_id.full_name()}#={dependency_id.version}")
+                    continue
+                elif remote_resolving:
+                    with stdlib.log.pushlog():
+                        dependency_fn = _solve_remotely(dependency)
```

Comment:
The `_fn` suffix seems weird to me, maybe this variable could be renamed to something clearer